packages/react-components/src/WithWalletConnector.ts

```diff
@@ -1,56 +1,96 @@
 import { Component } from 'react';
-import { SignClientTypes } from '@walletconnect/types';
-import {
-    BrowserWalletConnector,
-    Network,
-    WalletConnectConnector,
-    WalletConnection,
-    WalletConnectionDelegate,
-    WalletConnector,
-} from '@concordium/wallet-connectors';
+import { Network, WalletConnection, WalletConnectionDelegate, WalletConnector } from '@concordium/wallet-connectors';
 
-export type ConnectorType = 'BrowserWallet' | 'WalletConnect';
+/**
+ * Activation/deactivation controller of a given connector type.
+ */
+export interface ConnectorType {
+    /**
+     * Called when the connection type is being activated.
+     * The connector instance returned by this method becomes the new {@link State.activeConnector activeConnector}.
+     * @param component The component in which the instance is being activated.
+     *                  This object doubles as the delegate to pass to new connector instances.
+     * @param network The network to pass to new connector instances.
+     */
+    activate(component: WithWalletConnector, network: Network): Promise<WalletConnector>;
+
+    /**
+     * Called from {@link WithWalletConnector} when the connection type is being deactivated,
+     * i.e. right after {@link State.activeConnector activeConnector} has been unset from this value.
+     * @param component The component in which the instance is being deactivated.
+     * @param connector The connector to deactivate.
+     */
+    deactivate(component: WithWalletConnector, connector: WalletConnector): Promise<void>;
+}
+
+/**
+ * Produce a {@link ConnectorType} that creates a new connector instance on activation
+ * and disconnects the existing one on deactivation.
+ * This is the simplest connection type and should be used unless there's a reason not to.
+ * @param create Factory function for creating new connector instances.
+ */
+export function ephemeralConnectorType(create: (c: WithWalletConnector, n: Network) => Promise<WalletConnector>) {
+    return {
+        activate: create,
+        deactivate: (w: WithWalletConnector, c: WalletConnector) => c.disconnect(),
```

Comment:
It would if I didn't use the parameter after it. There's nothing special about `_` in JS/TS so this just names the variable `_`. You can therefore not use it more than once, so I would prefer not to use it.

Reply:
I know it's just a name, I swear I've had lint errors (or maybe typescript compiler warning "no unused variables") based on this. Maybe I'm wrong :stuck_out_tongue:

Reply:
Bad linter rule IMHO ;) Maybe it was because it couldn't figure out that you were implementing an interface? :shrug: